src/main/java/com/minecolonies/coremod/commands/AbstractCitizensCommands.java

```diff
@@ -0,0 +1,165 @@
+ */
+public abstract class AbstractCitizensCommands extends AbstractSingleCommand
+{
+    private static final String NO_ARGUMENTS                    = "Please define a valid citizen and/or colony";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public AbstractCitizensCommands(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <CitizenId>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        if(args.length == 0)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NO_ARGUMENTS));
+            return;
+        }
+
+        boolean firstArgumentColonyId = true;
+        int colonyId = -1;
+        if(args.length >= 2)
+        {
+            colonyId = getIthArgument(args, 0, -1);
+            if(colonyId == -1)
+            {
+                firstArgumentColonyId = false;
+            }
+        }
+
+        final Colony colony;
+        if(sender instanceof EntityPlayer && colonyId == -1)
+        {
+            IColony tempColony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
+            if (tempColony != null)
+            {
+                    colonyId = tempColony.getID();
+            }
+        }
+
+        colony = ColonyManager.getColony(colonyId);
+
+        if(colony == null)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NO_ARGUMENTS));
+            return;
+        }
+
+        if(sender instanceof EntityPlayer)
+        {
+            final EntityPlayer player = (EntityPlayer) sender;
+            if (!canPlayerUseCommand(player, getCommand(), colonyId))
+            {
+                sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NOT_PERMITTED));
+                return;
+            }
+        }
+
+        int citizenId = getValidCitizenId(colony, firstArgumentColonyId, args);
+
+        if(citizenId == -1)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NO_ARGUMENTS));
+            return;
+        }
+
+        executeSpecializedCode(server, sender, colony, citizenId);
+    }
+
+    /**
+     * Get a valid citizenid from the arguments.
+     * @param colony the colony.
+     * @param firstArgumentColonyId to define the offset.
+     * @param args the arguments.
+     * @return the valid id or -1 if not found.
+     */
+    private static int getValidCitizenId(final Colony colony, final boolean firstArgumentColonyId, final String...args)
+    {
+        int citizenId = getIthArgument(args, 1, -1);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'citizenId' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)